pwr.md

```diff
@@ -0,0 +1,144 @@
+* `hwp_enabled`             # HWP enabled flag
+* `base_freq`               # base frequency
+* `all_core_turbo_freq`     # all core turbo frequency
+* `highest_freq`            # highest available frequency
+* `lowest_freq`             # lowest active frequency
+* `uncore_hw_max`           # max available uncore frequency
+* `uncore_hw_min`           # min available uncore frequency
+* `uncore_freq`             # current uncore frequency
+* `uncore_max_freq`         # max desired uncore frequency
+* `uncore_min_freq`         # min desired uncore frequency
+
+### core
+* `core_id`                 # core id number
+* `cpu`                     # this cores cpu object
+* `thread_siblings`         # list of thread siblings
+* `high_prioirty`           # high/normal priority
+* `base_freq`               # base frequency [2300Mhz]
+* `sst_bf_base_freq`        # priority based frequency [2100Mhz/2700Mhz]
+* `all_core_turbo_freq`     # all core turbo frequency [2800Mhz]
+* `highest_freq`            # highest frequency available [3900Mhz]
+* `lowest_freq`             # lowest active frequency [800Mhz]
+* `curr_freq`               # current core frequency
+* `min_freq`                # desired low frequency
+* `max_freq`                # desired high frequency
+* `epp`                     # energy performance preference
+
+The **core** object attributes which can be adjusted are `max_freq`, `min_freq` and `epp`, min and max set the desired core frequency at which an application wants the core to operate within. 
+>Available frequencies are between `lowest_freq` and `highest_freq`, with some defined freqeuncies being `base_freq`/`sst_bf_base_freq`/`all_core_turbo_freq` if configuration allows.
+
+`epp` sets the desired performance profile for the core.
+>Available profiles are `default`, `performance`, `balance_performance`, `balance_power` and `power`.
+
+The **cpu** object attributes which can be adjusted by the application are `uncore_max_freq` and `uncore_min_freq`, these set the desired uncore frequency at which an application wants the uncore to operate within.
+>Available frequencies are between `uncore_hw_max` and `uncore_hw_min`.
+
+```python
+core.min_freq = core.lowest_freq
+core.max_freq = core.highest_freq
+
+cpu.uncore_max_freq = cpu.uncore_hw_max
+cpu.uncore_min_freq = cpu.uncore_hw_min
+```
+
+### Committing 
+Modification of the power settings of a system is done by altering the core or CPU characteristics, as shown above, 
+then finalizing with the `commit()` function call, this will commit your changes and write to relevant MSRs or sysfs entries.
+```python 
+cores = pwr.get_cores()
+
+for core in cores:
+    core.min_freq = core.base_freq
+    core.max_freq = core.highest_freq
+    core.commit()
+```
+
+
+### Pre-set Profiles
+When an application is modifying the desired min and max core frequencies, pre-set configurations can also be applied, these will overwrite current configurations and commit the pre-sets.  
```

Comment:
I would add a note here that the MHz values are based on a specific SKU and can be different for other SKU's.

Reply:
Oh, you _do_ have a note, but it's way below this :)

Perhaps a way around this would be to suppose an imaginary CPU with certain values, and go off that:
```
Given a CPU with min freq at 800, base freq at 2300, max freq at 3900, the profiles will do the following...
```